jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ResourceHandler.java

```diff
@@ -223,861 +138,154 @@ public Request.Processor handle(Request request) throws Exception
-
-                    if (!match)
-                    {
-                        Response.writeError(request, response, callback, HttpStatus.PRECONDITION_FAILED_412);
-                        return true;
-                    }
-                }
-
-                if (ifnm != null && etag != null)
-                {
-                    // Handle special case of exact match OR gzip exact match
-                    if (CompressedContentFormat.tagEquals(etag, ifnm) && ifnm.indexOf(',') < 0)
-                    {
-                        Response.writeError(request, response, callback, HttpStatus.NOT_MODIFIED_304);
-                        return true;
-                    }
-
-                    // Handle list of tags
-                    QuotedCSV quoted = new QuotedCSV(true, ifnm);
-                    for (String tag : quoted)
-                    {
-                        if (CompressedContentFormat.tagEquals(etag, tag))
-                        {
-                            Response.writeError(request, response, callback, HttpStatus.NOT_MODIFIED_304);
-                            return true;
-                        }
-                    }
-
-                    // If etag requires content to be served, then do not check if-modified-since
-                    return false;
-                }
-            }
-
-            // Handle if modified since
-            if (ifms != null)
-            {
-                //Get jetty's Response impl
-                String mdlm = content.getLastModifiedValue();
-                if (ifms.equals(mdlm))
-                {
-                    Response.writeError(request, response, callback, HttpStatus.NOT_MODIFIED_304);
-                    return true;
-                }
-
-                long ifmsl = request.getHeaders().getDateField(HttpHeader.IF_MODIFIED_SINCE.asString());
-                if (ifmsl != -1 && Files.getLastModifiedTime(content.getPath()).toMillis() / 1000 <= ifmsl / 1000)
-                {
-                    Response.writeError(request, response, callback, HttpStatus.NOT_MODIFIED_304);
-                    return true;
-                }
-            }
-
-            // Parse the if[un]modified dates and compare to resource
-            if (ifums != -1 && Files.getLastModifiedTime(content.getPath()).toMillis() / 1000 > ifums / 1000)
-            {
-                Response.writeError(request, response, callback, HttpStatus.PRECONDITION_FAILED_412);
-                return true;
-            }
-        }
-        catch (IllegalArgumentException iae)
-        {
-            if (!response.isCommitted())
-                Response.writeError(request, response, callback, HttpStatus.BAD_REQUEST_400);
-            throw iae;
-        }
-
-        return false;
-    }
-
-    protected void sendWelcome(HttpContent content, String pathInContext, boolean endsWithSlash, Request request, Response response, Callback callback) throws Exception
+    public Resource getResourceBase()
```

Comment:
No that we have restored `Resource` I think we can probably rename this method everywhere back to `getBaseResource()`... but I'll do that in another PR.